### ISSUE TITLE
Unit/integration test environment - jasmine

### ISSUE THREAD

**Issue by zoran995** [TerriaJS/terriajs#6961]
There are some issues with the current testing approach applied to terriajs project, making tests both fragile and hard to write
- using an outdated testing library, `jasmine` latest release is 5.1 while terriajs is using 2.9. It becomes pretty hard to find proper documentation and solutions to issues that arise while writing tests.
- having API requests in tests, currently, there are over 60 requests to the external API systems which is pretty bad IMHO. Those request also slows down running the unit tests, and I am pretty sure those can be blocked with proper mocking and early return, or replaced with local data load requests.
- not using a modern testing library made for testing react, yeah `react-test-renderer` provides some features but it is forcing usage of dom API and parsing of react tree to find simple stuff making tests pretty fragile. i.e. searching for a rendered string, a developer will have to find an element in a dom tree (maybe even traverse through its children) and check the proper content, any future change to the tested part of code will break the used code.

**Comment by pjonsson** [comment]
It's not going to solve this issue, but I tried updating to Jasmine-core 3.x and put up #7378 to fix some type errors in the tests, and #7379 to fix promise rejections being detected as unhandled. There's also #7160 from back in May that fixes more type errors in the testsuite. These three PRs are still not enough to make the test suite pass with Jasmine-core 3, but at least some tests run, and then after passing ~180 tests things stop.

A change in Jasmine 3 is that it runs tests in random order by default, and enabling that option with the current version of Jasmine gives a lot of failures. I guess the Jasmine upgrade could disable the random order, a more recent Jasmine version with random order disabled is no worse than the current situation.

There's also a copy of jasmine 2.99 in `wwwroot/third_party/jasmine-2.99.0` which I'm not sure how it is invoked, and if that version needs to be kept in lock-step with the version in `package.json`. It would be easier if that copy of Jasmine could be removed, and whatever Jasmine version that is installed in `node_modules` was used instead.

If we upgrade everything to the latest version, we can get karma-jasmine to use Jasmine 4.x out of the box. Some Jasmine maintainer commented that he didn't see anything in karma-jasmine that was affected by the Jasmine 5.x changes, so it's possible to add a clause to resolutions to force Jasmine 5.x.

I think not being able to use build-specs to get more easily readable backtraces and running the test suite without failures (#7288) makes progressing on this stuff harder, too.

**Comment by zoran995** [comment]
@pjonsson, we need to discuss internally about how to handle the test suite upgrade and the best path forward. Karma is also deprecated. We will post more details after evaluating possible options, and until then, I don't believe there is a point in investing time into upgrading the `jasmine` as we might even decide to migrate to another test runner.

**Comment by pjonsson** [comment]
> I don't believe there is a point in investing time into upgrading the `jasmine` as we might even decide to migrate to another test runner.

Agreed @zoran995. And to avoid misunderstandings I'll explicitly mention that the PRs I linked are fixing problems with the tests themselves, not with Jasmine/Karma.

**Comment by zoran995** [comment]
We decided to upgrade Jasmine to the latest version and use jasmine-browser-runner to minimise breaking changes for now, so we can prioritise other, more urgent things. 
We also experimented with using Vitest, and have a working POC, but there is not much point in doing that without migrating the build process from webpack to Vite, as otherwise we would have two completely separated build processes to maintain one for tests and one for dev/production environment. There are no significant performance improvements with Vitest over Jasmine v6 when running everything in the browser.
We might revisit this decision in the long run as part of making TerriaJS more modular and usable outside of TerriaMap.
